cli/auth_nkey_command.go

```diff
@@ -256,3 +272,106 @@ func (c *authNKCommand) verifyAction(_ *fisk.ParseContext) error {
 
 	return nil
 }
+func (c *authNKCommand) sealAction(_ *fisk.ParseContext) error {
+	var err error
+	var kp nkeys.KeyPair
+
+	keyData, err := c.readKeyFile(c.keyFile)
+	if err != nil {
+		return err
+	}
+
+	// try it as public, then as seed
+	kp, err = nkeys.FromPublicKey(string(keyData))
+	if errors.Is(err, nkeys.ErrInvalidPublicKey) {
+		kp, err = nkeys.FromSeed(keyData)
+	}
+	if err != nil {
+		return err
+	}
+
+	content, err := os.ReadFile(c.dataFile)
+	if err != nil {
+		return err
+	}
+
+	if nkeys.IsValidPublicCurveKey(c.counterpartKey) {
```

Comment:
thnk its better to change this to

```
if !nkeys.IsValidPublicCurveKey(c.counterpartKey) {
  return fmt.Errorf("some nice error")
}
```

and unindent all the rest so you handle all the validation upfront, its also just nice to not wrap the meat of a function inside a conditional - obviously a minor nit.